---
title: ""
output: html_document
author: Kelsey Keith
date: '2022-03-25'
---

```{r setup, include=FALSE}
### essential libraries
library(magrittr)
library(dplyr)
library(readr)
library(gt)

knitr::opts_chunk$set(echo = TRUE)
```

## Count Biospecimens 

In the Pediatric Molecular Open Targets high-level documentation <https://github.com/PediatricOpenTargets/documentation>, we have a table with the number of samples per cohort for the Datasets section so that people are aware of the general number of samples they can access. Data can be accessed through the `OpenPedCan-analysis` repository <https://github.com/PediatricOpenTargets/OpenPedCan-analysis>, although for privacy reasons not all data, especially raw data is accessible publically. This script calculates and outputs the table.

<br>

#### Install Required Packages

If you don't have all the packages listed below, please uncomment `install.packages()` in the chunk below and install them.

```{r}
# install.packages(c('magrittr', 'dplyr', 'readr', 'gt', 'webshot2'))
```

<br>

#### Data Retrieval 

The `OpenPedCan-analysis` repository is a submodule of this repository; git submodules allow you to reference other git projects while still keeping them separate. In order to run this script, you first need to make sure the submodule is set up and the data is downloaded within it. 

1. First, clone this repository

```{bash, eval = F}
git clone https://github.com/PediatricOpenTargets/documentation
```

2. When a repository with a submodule is cloned, the information inside the submodule does not come with it. Each user must initialize and update the submodule individually.

```{bash, eval = F}
cd documentation
git submodule init
git submodule update
```

3. Finally, the `OpenPedCan-analysis` repository does not contain all the data within it, as it is too large to be stored on GitHub. However, it provides a script with which to download the data, which you can read more about [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis#data-access-via-download-script). Run the `download-data.sh` script to download the data needed to run this script.

```{bash, eval = F}
cd OpenPedCan-analysis
bash download-data.sh
```

<br><br>

### Calculate Biospecimen Count Table

We want to have a table to display in the README with the number of biospecimens used on the Pediatric Molecular Targets site and that may potentially be available for researchers for further use.

---

#### Read In Data

Read in tables. The file `histologies.tsv` contains the cancer assignment for each biospecimen in the `cancer_group` column. Some participants have more than one biospecimen, but since many analyses require that all samples be independent, for participants with multiple samples, a single sample has been selected. The `independent-specimen` lists specify the selected sample for all participants. You can read more about the logica and selection process behind the `independent-specimen` lists [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/independent-samples).

```{r}
### sample metadata file, including the cancer_group
read_tsv('../OpenPedCan-analysis/data/histologies.tsv') -> histologies

### independent specimen lists
# RNA-seq
read_tsv('../OpenPedCan-analysis/data/independent-specimens.rnaseqpanel.primary.eachcohort.tsv') -> rna_primary
read_tsv('../OpenPedCan-analysis/data/independent-specimens.rnaseqpanel.relapse.eachcohort.tsv') -> rna_relapse
# DNA
read_tsv('../OpenPedCan-analysis/data/independent-specimens.wgswxspanel.primary.eachcohort.prefer.wgs.tsv') -> dna_primary
read_tsv('../OpenPedCan-analysis/data/independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wgs.tsv') -> dna_relapse
# methylation
read_tsv('../OpenPedCan-analysis/data/independent-specimens.methyl.primary.eachcohort.tsv') -> methyl_primary
read_tsv('../OpenPedCan-analysis/data/independent-specimens.methyl.relapse.eachcohort.tsv') -> methyl_relapse
```

<br>

#### Count DNA Biospecimens

Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort

```{r}
### Count DNA samples
histologies %>% 
# get unique combinations of patients, cohorts, diseases (cancer_group), 
# types of data (experimental_strategy), and tumor descriptions
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>% 
  inner_join(bind_rows(dna_primary, dna_relapse)) %>% 
  filter(!is.na(cohort), !is.na(cancer_group)) %>%
# add stage column to specify whether the tumor is primary or relapse in keeping
# with the language on the Pediatric Molecular Targets site
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_primary$tumor_descriptor) ~ 'primary',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',
                           TRUE ~ NA_character_),
# Don't distinguish different sequencing methods on the Pediatric Molecular 
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all
# samples as DNA
         sequencing_type = 'DNA',
         experimental_strategy = case_when(experimental_strategy %in% 
                                            c('WGS', 'WXS') ~ 'WGS/WXS', 
                                          experimental_strategy == 
                                            'Targeted Sequencing' ~ 'SNV Panel',
                                          TRUE ~ experimental_strategy)) %>% 
  count(cohort, stage, sequencing_type, 
        experimental_strategy, cancer_group) %>%
  filter(n >= 3) -> dna_counts
```

Check that there is no missing information (`NA`) in the DNA count table.

```{r}
if ( any(is.na(dna_counts$stage)) ) {
  warning('Missing stage information, is.na(dna_counts$stage) is true.')
} else if ( any(is.na(dna_counts$sequencing_type)) ) {
  warning('Missing sequencing type information, is.na(dna_counts$sequencing_type) is true.')
} else if ( any(is.na(dna_counts$n)) ) {
  warning('Missing biospecimen count information, is.na(dna_counts$n) is true.')
} else if ( any(is.na(dna_counts$cancer_group)) ) {
  warning('Missing cancer_group count information, is.na(dna_counts$cancer_group) is true.')
} 
```

<br>

#### Count RNA Biospecimens

Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort

```{r}
### Count DNA samples
histologies %>% 
# get unique combinations of patients, cohorts, diseases (cancer_group), 
# types of data (experimental_strategy), and tumor descriptions
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>%
  filter(Kids_First_Biospecimen_ID %in% rna_primary$Kids_First_Biospecimen_ID | 
           Kids_First_Biospecimen_ID %in% rna_relapse$Kids_First_Biospecimen_ID |
           cohort == 'GTEx') %>% 
  filter(!is.na(cohort), !is.na(cancer_group) | sample_type == 'Normal') %>%
# add stage column to specify whether the tumor is primary or relapse in keeping
# with the language on the Pediatric Molecular Targets site
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_primary$tumor_descriptor) ~ 'primary',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',
                           TRUE ~ NA_character_),
# Don't distinguish different sequencing methods on the Pediatric Molecular 
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all
# samples as DNA
         sequencing_type = 'RNA',
         experimental_strategy = case_when(experimental_strategy == 
                                             'Targeted Sequencing' ~ 
                                             'Fusion Panel',
                                           TRUE ~ experimental_strategy)) %>%
  count(cohort, stage, sequencing_type, 
        experimental_strategy, cancer_group) %>%
  filter(n >= 3) -> rna_counts
```

Check that there is no missing information (`NA`) in the RNA count table.

```{r}
if ( any(is.na(rna_counts$stage)) ) {
  warning('Missing stage information, is.na(rna_counts$stage) is true.')
} else if ( any(is.na(rna_counts$sequencing_type)) ) {
  warning('Missing sequencing type information, is.na(rna_counts$sequencing_type) is true.')
} else if ( any(is.na(rna_counts$n)) ) {
  warning('Missing biospecimen count information, is.na(rna_counts$n) is true.')
} else if ( any(is.na(rna_counts$cancer_group)) ) {
  warning('Missing cancer_group count information, is.na(rna_counts$cancer_group) is true.')
} 
```

<br>

#### Count Methylation Biospecimens

Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort

```{r}
### Count DNA samples
histologies %>% 
# get unique combinations of patients, cohorts, diseases (cancer_group), 
# types of data (experimental_strategy), and tumor descriptions
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>%
  inner_join(rbind(methyl_primary, methyl_relapse)) %>% 
  filter(!is.na(cohort), !is.na(cancer_group)) %>%
# add stage column to specify whether the tumor is primary or relapse in keeping
# with the language on the Pediatric Molecular Targets site
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_primary$tumor_descriptor) ~ 'primary',
                           sample_type == 'Tumor' & tumor_descriptor %in% 
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',
                           TRUE ~ NA_character_),
# Don't distinguish different sequencing methods on the Pediatric Molecular 
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all
# samples as DNA
         sequencing_type = 'Methylation Array') %>% 
  count(cohort, stage, sequencing_type, 
        experimental_strategy, cancer_group) -> methyl_counts
```

Check that there is no missing information (`NA`) in the methylation count table.

```{r}
if ( any(is.na(methyl_counts$stage)) ) {
  warning('Missing stage information, is.na(methyl_counts$stage) is true.')
} else if ( any(is.na(methyl_counts$sequencing_type)) ) {
  warning('Missing sequencing type information, is.na(methyl_counts$sequencing_type) is true.')
} else if ( any(is.na(methyl_counts$n)) ) {
  warning('Missing biospecimen count information, is.na(methyl_counts$n) is true.')
} else if ( any(is.na(methyl_counts$cancer_group)) ) {
  warning('Missing cancer_group count information, is.na(methyl_counts$cancer_group) is true.')
} 
```

<br>

### Combine Count Tables

Combine the count tables, organize them for display, and save.

```{r}
rbind(dna_counts, rna_counts, methyl_counts) %>%
# change cohort names to be in line with the language on the Pediatric 
# Molecular Targets website
  mutate(cohort = case_when(cohort == 'DGD' ~ 'CHOP P30 Panel',
                            cohort == 'TARGET' & 
                              experimental_strategy == 'Targeted Sequencing' ~ 
                              'TARGET Panel',
                            TRUE ~ cohort)) -> dataset_disease_counts
```

#### Per Dataset and Disease Counts

```{r}
dataset_disease_counts %>% 
# change column names to be in line with the language on the Pediatric 
# Molecular Targets website and retain desired columns
  select(Dataset = cohort, `Sequencing Type` = sequencing_type,
         `Experimental Strategy` = experimental_strategy,
        `Tumor Stage` = stage, Disease = cancer_group, 
        `Number of Biospecimens` = n) %>%
  arrange(Dataset, Disease, `Sequencing Type`, `Experimental Strategy`,
          `Tumor Stage`) -> dataset_disease_counts_filtered

# save as a plain text table
write_csv(dataset_disease_counts_filtered, 
          'biospecimen_dataset_disease_counts.csv')
write_csv(dataset_disease_counts_filtered, 
          '../disease_subject_counts.tsv')
```

#### Per Dataset Counts 

Save the the per dataset counts, both as a CSV and, primarily, as an image to display in the repository README. Troubleshooting: If you get a Google Chrome timeout error, please knit the document again and it should be resolved.

```{r}
dataset_disease_counts %>%
  group_by(cohort, stage, sequencing_type, experimental_strategy) %>%
  summarize(biospecimen_count = sum(n)) %>%
  ungroup() %>%
# change column names to be in line with the language on the Pediatric 
# Molecular Targets website and retain desired columns
  select(Dataset = cohort, `Sequencing Type` = sequencing_type,
         `Experimental Strategy` = experimental_strategy,
          `Tumor Stage` = stage, `Number of Biospecimens` = biospecimen_count) %>%
  arrange(Dataset, `Sequencing Type`, `Tumor Stage`)  -> dataset_counts

# save as a plain text table
write_csv(dataset_counts, 'biospecimen_dataset_counts.csv')

# save a pretty formatted table as an image so it can be automatically plugged 
# into the main repository README
dataset_counts %>%
  gt(rowname_col = F) %>% 
  opt_row_striping() %>%
  gtsave('../figures/biospecimen_dataset_counts.png')
```

<br><br>

