{\rtf1\ansi\ansicpg1252\cocoartf2708
\cocoatextscaling0\cocoaplatform0{\fonttbl\f0\fnil\fcharset0 HelveticaNeue;\f1\fnil\fcharset0 HelveticaNeue-Medium;\f2\fmodern\fcharset0 Courier;
\f3\fnil\fcharset0 Menlo-Regular;\f4\fnil\fcharset0 AppleSymbols;}
{\colortbl;\red255\green255\blue255;\red38\green38\blue38;\red41\green101\blue168;\red0\green0\blue0;
\red242\green242\blue242;\red255\green255\blue255;}
{\*\expandedcolortbl;;\cssrgb\c20000\c20000\c20000;\cssrgb\c20000\c47843\c71765;\cssrgb\c0\c0\c0\c3922;
\cssrgb\c96078\c96078\c96078;\cssrgb\c100000\c100000\c100000;}
{\*\listtable{\list\listtemplateid1\listhybrid{\listlevel\levelnfc0\levelnfcn0\leveljc0\leveljcn0\levelfollow0\levelstartat1\levelspace360\levelindent0{\*\levelmarker \{decimal\}}{\leveltext\leveltemplateid1\'01\'00;}{\levelnumbers\'01;}\fi-360\li720\lin720 }{\listname ;}\listid1}
{\list\listtemplateid2\listhybrid{\listlevel\levelnfc0\levelnfcn0\leveljc0\leveljcn0\levelfollow0\levelstartat2\levelspace360\levelindent0{\*\levelmarker \{decimal\}}{\leveltext\leveltemplateid101\'01\'00;}{\levelnumbers\'01;}\fi-360\li720\lin720 }{\listname ;}\listid2}
{\list\listtemplateid3\listhybrid{\listlevel\levelnfc0\levelnfcn0\leveljc0\leveljcn0\levelfollow0\levelstartat3\levelspace360\levelindent0{\*\levelmarker \{decimal\}}{\leveltext\leveltemplateid201\'01\'00;}{\levelnumbers\'01;}\fi-360\li720\lin720 }{\listname ;}\listid3}}
{\*\listoverridetable{\listoverride\listid1\listoverridecount0\ls1}{\listoverride\listid2\listoverridecount0\ls2}{\listoverride\listid3\listoverridecount0\ls3}}
{\info
{\author Kelsey Keith}}\margl1440\margr1440\vieww20380\viewh15920\viewkind0
\deftab720
\pard\pardeftab720\partightenfactor0

\f0\fs28 \cf2 \expnd0\expndtw0\kerning0
\outl0\strokewidth0 \strokec2 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs60 \cf2 Count Biospecimens\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 In the Pediatric Molecular Open Targets high-level documentation {\field{\*\fldinst{HYPERLINK "https://github.com/PediatricOpenTargets/documentation"}}{\fldrslt \cf3 \strokec3 https://github.com/PediatricOpenTargets/documentation}}, we have a table with the number of samples per cohort for the Datasets section so that people are aware of the general number of samples they can access. Data can be accessed through the 
\f2\fs25\fsmilli12600 \cb4 OpenPedCan-analysis
\f0\fs28 \cb1  repository {\field{\*\fldinst{HYPERLINK "https://github.com/PediatricOpenTargets/OpenPedCan-analysis"}}{\fldrslt \cf3 \strokec3 https://github.com/PediatricOpenTargets/OpenPedCan-analysis}}, although for privacy reasons not all data, especially raw data is accessible publically. This script calculates and outputs the table.\
\
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Install Required Packages\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 If you don\'92t have all the packages listed below, please uncomment 
\f2\fs25\fsmilli12600 \cb4 install.packages()
\f0\fs28 \cb1  in the chunk below and install them.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 # install.packages(c('magrittr', 'dplyr', 'readr', 'gt', 'webshot2'))\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Data Retrieval\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 The 
\f2\fs25\fsmilli12600 \cb4 OpenPedCan-analysis
\f0\fs28 \cb1  repository is a submodule of this repository; git submodules allow you to reference other git projects while still keeping them separate. In order to run this script, you first need to make sure the submodule is set up and the data is downloaded within it.\
\pard\tx220\tx720\pardeftab720\li720\fi-720\partightenfactor0
\ls1\ilvl0\cf2 \kerning1\expnd0\expndtw0 \outl0\strokewidth0 {\listtext	1	}\expnd0\expndtw0\kerning0
\outl0\strokewidth0 \strokec2 First, clone this repository\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 git clone https://github.com/PediatricOpenTargets/documentation\
\pard\tx220\tx720\pardeftab720\li720\fi-720\partightenfactor0
\ls2\ilvl0
\f0\fs28 \cf2 \cb1 \kerning1\expnd0\expndtw0 \outl0\strokewidth0 {\listtext	2	}\expnd0\expndtw0\kerning0
\outl0\strokewidth0 \strokec2 When a repository with a submodule is cloned, the information inside the submodule does not come with it. Each user must initialize and update the submodule individually.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 cd documentation\
git submodule init\
git submodule update\
\pard\tx220\tx720\pardeftab720\li720\fi-720\partightenfactor0
\ls3\ilvl0
\f0\fs28 \cf2 \cb1 \kerning1\expnd0\expndtw0 \outl0\strokewidth0 {\listtext	3	}\expnd0\expndtw0\kerning0
\outl0\strokewidth0 \strokec2 Finally, the 
\f2\fs25\fsmilli12600 \cb4 OpenPedCan-analysis
\f0\fs28 \cb1  repository does not contain all the data within it, as it is too large to be stored on GitHub. However, it provides a script with which to download the data, which you can read more about {\field{\*\fldinst{HYPERLINK "https://github.com/PediatricOpenTargets/OpenPedCan-analysis#data-access-via-download-script"}}{\fldrslt \cf3 \strokec3 here}}. Run the 
\f2\fs25\fsmilli12600 \cb4 download-data.sh
\f0\fs28 \cb1  script to download the data needed to run this script.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 cd OpenPedCan-analysis\
bash download-data.sh\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \uc0\u8232 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs48 \cf2 Calculate Biospecimen Count Table\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 We want to have a table to display in the README with the number of biospecimens used on the Pediatric Molecular Targets site and that may potentially be available for researchers for further use.\
\pard\pardeftab720\partightenfactor0
\cf2 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Read In Data\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 Read in tables. The file 
\f2\fs25\fsmilli12600 \cb4 histologies.tsv
\f0\fs28 \cb1  contains the cancer assignment for each biospecimen in the 
\f2\fs25\fsmilli12600 \cb4 cancer_group
\f0\fs28 \cb1  column. Some participants have more than one biospecimen, but since many analyses require that all samples be independent, for participants with multiple samples, a single sample has been selected. The 
\f2\fs25\fsmilli12600 \cb4 independent-specimen
\f0\fs28 \cb1  lists specify the selected sample for all participants. You can read more about the logica and selection process behind the 
\f2\fs25\fsmilli12600 \cb4 independent-specimen
\f0\fs28 \cb1  lists {\field{\*\fldinst{HYPERLINK "https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/independent-samples"}}{\fldrslt \cf3 \strokec3 here}}.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 ### sample metadata file, including the cancer_group\
read_tsv('../OpenPedCan-analysis/data/histologies.tsv') -> histologies\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Warning: One or more parsing issues, call `problems()` on your data frame for details,\
## e.g.:\
##   dat <- vroom(...)\
##   problems(dat)\
## Rows: 43729 Columns: 55\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (38): Kids_First_Biospecimen_ID, sample_id, aliquot_id, Kids_First_Parti...\
## dbl (17): age_at_diagnosis_days, OS_days, EFS_days, age_last_update_days, no...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 ### independent specimen lists\
# RNA-seq\
read_tsv('../OpenPedCan-analysis/data/independent-specimens.rnaseqpanel.primary.eachcohort.tsv') -> rna_primary\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 12558 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 read_tsv('../OpenPedCan-analysis/data/independent-specimens.rnaseqpanel.relapse.eachcohort.tsv') -> rna_relapse\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 865 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 # DNA\
read_tsv('../OpenPedCan-analysis/data/independent-specimens.wgswxspanel.primary.eachcohort.prefer.wgs.tsv') -> dna_primary\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 3462 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 read_tsv('../OpenPedCan-analysis/data/independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wgs.tsv') -> dna_relapse\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 475 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 # methylation\
read_tsv('../OpenPedCan-analysis/data/independent-specimens.methyl.primary.eachcohort.tsv') -> methyl_primary\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 1750 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\partightenfactor0
\cf2 \cb5 read_tsv('../OpenPedCan-analysis/data/independent-specimens.methyl.relapse.eachcohort.tsv') -> methyl_relapse\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Rows: 484 Columns: 6\
## 
\f3 \uc0\u9472 \u9472 
\f2  Column specification 
\f3 \uc0\u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 \u9472 
\f2 \
## Delimiter: "\\t"\
## chr (6): Kids_First_Participant_ID, Kids_First_Biospecimen_ID, cohort, cance...\
## \
## 
\f4 \uc0\u8505 
\f2  Use `spec()` to retrieve the full column specification for this data.\
## 
\f4 \uc0\u8505 
\f2  Specify the column types or set `show_col_types = FALSE` to quiet this message.\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Count DNA Biospecimens\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 ### Count DNA samples\
histologies %>% \
# get unique combinations of patients, cohorts, diseases (cancer_group), \
# types of data (experimental_strategy), and tumor descriptions\
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,\
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>% \
  inner_join(bind_rows(dna_primary, dna_relapse)) %>% \
  filter(!is.na(cohort), !is.na(cancer_group)) %>%\
# add stage column to specify whether the tumor is primary or relapse in keeping\
# with the language on the Pediatric Molecular Targets site\
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_primary$tumor_descriptor) ~ 'primary',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',\
                           TRUE ~ NA_character_),\
# Don't distinguish different sequencing methods on the Pediatric Molecular \
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all\
# samples as DNA\
         sequencing_type = 'DNA',\
         experimental_strategy = case_when(experimental_strategy %in% \
                                            c('WGS', 'WXS') ~ 'WGS/WXS', \
                                          experimental_strategy == \
                                            'Targeted Sequencing' ~ 'SNV Panel',\
                                          TRUE ~ experimental_strategy)) %>% \
  count(cohort, stage, sequencing_type, \
        experimental_strategy, cancer_group) %>%\
  filter(n >= 3) -> dna_counts\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Joining with `by = join_by(Kids_First_Participant_ID,\
## Kids_First_Biospecimen_ID, cohort, cancer_group, experimental_strategy,\
## tumor_descriptor)`\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 Check that there is no missing information (
\f2\fs25\fsmilli12600 \cb4 NA
\f0\fs28 \cb1 ) in the DNA count table.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 if ( any(is.na(dna_counts$stage)) ) \{\
  warning('Missing stage information, is.na(dna_counts$stage) is true.')\
\} else if ( any(is.na(dna_counts$sequencing_type)) ) \{\
  warning('Missing sequencing type information, is.na(dna_counts$sequencing_type) is true.')\
\} else if ( any(is.na(dna_counts$n)) ) \{\
  warning('Missing biospecimen count information, is.na(dna_counts$n) is true.')\
\} else if ( any(is.na(dna_counts$cancer_group)) ) \{\
  warning('Missing cancer_group count information, is.na(dna_counts$cancer_group) is true.')\
\} \
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Count RNA Biospecimens\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 ### Count DNA samples\
histologies %>% \
# get unique combinations of patients, cohorts, diseases (cancer_group), \
# types of data (experimental_strategy), and tumor descriptions\
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,\
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>%\
  filter(Kids_First_Biospecimen_ID %in% rna_primary$Kids_First_Biospecimen_ID | \
           Kids_First_Biospecimen_ID %in% rna_relapse$Kids_First_Biospecimen_ID |\
           cohort == 'GTEx') %>% \
  filter(!is.na(cohort), !is.na(cancer_group) | sample_type == 'Normal') %>%\
# add stage column to specify whether the tumor is primary or relapse in keeping\
# with the language on the Pediatric Molecular Targets site\
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_primary$tumor_descriptor) ~ 'primary',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',\
                           TRUE ~ NA_character_),\
# Don't distinguish different sequencing methods on the Pediatric Molecular \
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all\
# samples as DNA\
         sequencing_type = 'RNA',\
         experimental_strategy = case_when(experimental_strategy == \
                                             'Targeted Sequencing' ~ \
                                             'Fusion Panel',\
                                           TRUE ~ experimental_strategy)) %>%\
  count(cohort, stage, sequencing_type, \
        experimental_strategy, cancer_group) %>%\
  filter(n >= 3) -> rna_counts\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 Check that there is no missing information (
\f2\fs25\fsmilli12600 \cb4 NA
\f0\fs28 \cb1 ) in the RNA count table.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 if ( any(is.na(rna_counts$stage)) ) \{\
  warning('Missing stage information, is.na(rna_counts$stage) is true.')\
\} else if ( any(is.na(rna_counts$sequencing_type)) ) \{\
  warning('Missing sequencing type information, is.na(rna_counts$sequencing_type) is true.')\
\} else if ( any(is.na(rna_counts$n)) ) \{\
  warning('Missing biospecimen count information, is.na(rna_counts$n) is true.')\
\} else if ( any(is.na(rna_counts$cancer_group)) ) \{\
  warning('Missing cancer_group count information, is.na(rna_counts$cancer_group) is true.')\
\} \
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Warning: Missing cancer_group count information, is.na(rna_counts$cancer_group)\
## is true.\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 Count Methylation Biospecimens\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 Count the number of biospecimens per diease per sequencing type (aka experimental strategy) per cohort\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 ### Count DNA samples\
histologies %>% \
# get unique combinations of patients, cohorts, diseases (cancer_group), \
# types of data (experimental_strategy), and tumor descriptions\
  distinct(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_type,\
           cohort, cancer_group, experimental_strategy, tumor_descriptor) %>%\
  inner_join(rbind(methyl_primary, methyl_relapse)) %>% \
  filter(!is.na(cohort), !is.na(cancer_group)) %>%\
# add stage column to specify whether the tumor is primary or relapse in keeping\
# with the language on the Pediatric Molecular Targets site\
  mutate(stage = case_when(sample_type == 'Normal' ~ 'normal',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_primary$tumor_descriptor) ~ 'primary',\
                           sample_type == 'Tumor' & tumor_descriptor %in% \
                             unique(dna_relapse$tumor_descriptor) ~ 'relapse',\
                           TRUE ~ NA_character_),\
# Don't distinguish different sequencing methods on the Pediatric Molecular \
# Targets site, just the thing measured, SNV, CNV, Fusion, so just labeling all\
# samples as DNA\
         sequencing_type = 'Methylation Array') %>% \
  count(cohort, stage, sequencing_type, \
        experimental_strategy, cancer_group) -> methyl_counts\
\pard\pardeftab720\partightenfactor0
\cf2 \cb6 ## Joining with `by = join_by(Kids_First_Participant_ID,\
## Kids_First_Biospecimen_ID, cohort, cancer_group, experimental_strategy,\
## tumor_descriptor)`\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 Check that there is no missing information (
\f2\fs25\fsmilli12600 \cb4 NA
\f0\fs28 \cb1 ) in the methylation count table.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 if ( any(is.na(methyl_counts$stage)) ) \{\
  warning('Missing stage information, is.na(methyl_counts$stage) is true.')\
\} else if ( any(is.na(methyl_counts$sequencing_type)) ) \{\
  warning('Missing sequencing type information, is.na(methyl_counts$sequencing_type) is true.')\
\} else if ( any(is.na(methyl_counts$n)) ) \{\
  warning('Missing biospecimen count information, is.na(methyl_counts$n) is true.')\
\} else if ( any(is.na(methyl_counts$cancer_group)) ) \{\
  warning('Missing cancer_group count information, is.na(methyl_counts$cancer_group) is true.')\
\} \
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 \cb1 \
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs48 \cf2 Combine Count Tables\
\pard\pardeftab720\sa200\partightenfactor0

\f0\fs28 \cf2 Combine the count tables, organize them for display, and save.\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 rbind(dna_counts, rna_counts, methyl_counts) %>%\
# change cohort names to be in line with the language on the Pediatric \
# Molecular Targets website\
  mutate(cohort = case_when(cohort == 'DGD' ~ 'CHOP P30 Panel',\
                            cohort == 'TARGET' & \
                              experimental_strategy == 'Targeted Sequencing' ~ \
                              'TARGET Panel',\
                            TRUE ~ cohort)) -> dataset_disease_counts\
\pard\pardeftab720\sa200\partightenfactor0

\f1\fs36 \cf2 \cb1 Per Dataset and Disease Counts\
\pard\pardeftab720\partightenfactor0

\f2\fs26 \cf2 \cb5 dataset_disease_counts %>% \
# change column names to be in line with the language on the Pediatric \
# Molecular Targets website and retain desired columns\
  select(Dataset = cohort, `Sequencing Type` = sequencing_type,\
         `Experimental Strategy` = experimental_strategy,\
        `Tumor Stage` = stage, Disease = cancer_group, \
        `Number of Biospecimens` = n) %>%\
  arrange(Dataset, Disease, `Sequencing Type`, `Experimental Strategy`,\
          `Tumor Stage`) -> dataset_disease_counts_filtered\
\
# save as a plain text table\
write_csv(dataset_disease_counts_filtered, \
          'biospecimen_dataset_disease_counts.csv')\
write_csv(dataset_disease_counts_filtered, \
          '../disease_subject_counts.tsv')\
\pard\pardeftab720\partightenfactor0

\fs25\fsmilli12600 \cf2 \cb4 \
\
\pard\pardeftab720\sa200\partightenfactor0

\fs36 \cf2 Per Dataset Counts
\fs25\fsmilli12600 \
\pard\pardeftab720\sa200\partightenfactor0
\cf2 Save the the per dataset counts, both as a CSV and, primarily, as an\
image to display in the repository README. Troubleshooting: If you get a\
Google Chrome timeout error, please knit the document again and it\
should be resolved.\
\pard\pardeftab720\partightenfactor0

\fs26 \cf2 \cb5 dataset_disease_counts %>%\
  group_by(cohort, stage, sequencing_type, experimental_strategy) %>%\
  summarize(biospecimen_count = sum(n)) %>%\
  ungroup() %>%\
# change column names to be in line with the language on the Pediatric \
# Molecular Targets website and retain desired columns\
  select(Dataset = cohort, `Sequencing Type` = sequencing_type,\
         `Experimental Strategy` = experimental_strategy,\
          `Tumor Stage` = stage, `Number of Biospecimens` = biospecimen_count) %>%\
  arrange(Dataset, `Sequencing Type`, `Tumor Stage`)  -> dataset_counts
\fs25\fsmilli12600 \cb4 \
\pard\pardeftab720\partightenfactor0

\fs26 \cf2 \cb6 ## `summarise()` has grouped output by 'cohort', 'stage', 'sequencing_type'. You\
## can override using the `.groups` argument.
\fs25\fsmilli12600 \cb4 \
\pard\pardeftab720\partightenfactor0

\fs26 \cf2 \cb5 # save as a plain text table\
write_csv(dataset_counts, 'biospecimen_dataset_counts.csv')\
\
# save a pretty formatted table as an image so it can be automatically plugged \
# into the main repository README\
dataset_counts %>%\
  gt(rowname_col = F) %>% \
  opt_row_striping() %>%\
  gtsave('../figures/biospecimen_dataset_counts.png')
\fs25\fsmilli12600 \cb4 \
\pard\pardeftab720\sa200\partightenfactor0
\cf2 \cb1 \uc0\u8232 \cb4 \
\pard\pardeftab720\partightenfactor0
\cf2 \
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
\
}